dandi/dandiapi.py

```diff
@@ -325,24 +349,82 @@ class Version(APIBase):
     modified: datetime
 
 
-class RemoteDandiset(APIBase):
+class RemoteDandiset:
     """
     Representation of a Dandiset (as of a certain version) retrieved from the
     API
     """
 
-    client: "DandiAPIClient"
-    identifier: str
-    created: datetime
-    modified: datetime
-    #: The version in question of the Dandiset
-    version: Version
-    most_recent_published_version: Optional[Version]
-    draft_version: Version
+    def __init__(
+        self,
+        client: DandiAPIClient,
+        identifier: str,
+        version: Union[str, Version, None],
+        data: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        self.client = client
+        self.identifier = identifier
+        self._version_id: Optional[str]
+        self._version: Optional[Version]
+        if version is None:
+            self._version_id = None
+            self._version = None
+        elif isinstance(version, str):
+            self._version_id = version
+            self._version = None
+        else:
+            self._version_id = version.identifier
+            self._version = version
+        self._data = data
+
+    def _get_data(self) -> Dict[str, Any]:
+        if self._data is None:
+            self._data = self.client.get(f"/dandisets/{self.identifier}/")
+        return self._data
 
     @property
     def version_id(self) -> str:
-        return self.version.identifier
+        if self._version_id is None:
+            self._version_id = self.version.identifier
+        return self._version_id
+
+    @property
+    def version(self) -> Version:
+        """The version in question of the Dandiset"""
+        if self._version is None:
```

Comment:
should we may be default `version` to None in `__init__` if we can handle it?

Reply:
Done.